### PR TITLE
B/plugin location

### PR DESCRIPTION
**Pull request by m-kharbat**

### CODE REVIEW

File: app/tests/services/sql.tests.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  describe('SQL baseURI', function() {
+    it('base uri should remove admin path', inject(function(baseURI) {
+      //retrieve initial path
+      var initial_location = window.location.pathname;
+      //replace path with '/admin'
+      history.replaceState(  { test: "test" }, 'test', '/something/admin/');
+      expect(baseURI.getURI('/test')).toEqual(window.location.protocol + '//' + window.location.host + '/something/test');
+      //restore inital path
+      history.replaceState({ test: "test" }, 'test',initial_location);
```
Comment (bogensberger) [inline app/tests/services/sql.tests.js]: space after `,`

File: app/tests/services/sql.tests.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }));
+  });
+
+
+  describe('SQL baseURI', function() {
+    it('base uri should remove plugin path', inject(function(baseURI) {
+      //retrieve initial path
+      var initial_location = window.location.pathname;
+      //replace path with '_plugin/crate-admin/'
+      history.replaceState( { test: "test" }, 'test', '/something/_plugin/crate-admin/');
```
Comment (bogensberger) [inline app/tests/services/sql.tests.js]: remove space after `(`

File: app/tests/services/sql.tests.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      history.replaceState({ test: "test" }, 'test', initial_location);
+    }));
+  });
+
+  describe('SQL baseURI', function() {
+    it('base uri should remove admin path', inject(function(baseURI) {
+      //retrieve initial path
+      var initial_location = window.location.pathname;
+      //replace path with '/admin'
+      history.replaceState(  { test: "test" }, 'test', '/something/admin/');
```
Comment (bogensberger) [inline app/tests/services/sql.tests.js]: remove space after `(`

File: app/tests/services/sql.tests.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  beforeEach(function() {
+    angular.mock.inject(function($injector) {
+      baseURI = $injector.get('baseURI');
+    });
+  });
+
+  describe('SQL baseURI', function() {
+    it('base uri should keep path', inject(function(baseURI) {
+      //retrieve initial path
+      expect(baseURI.getURI('/test')).toEqual(window.location.protocol + '//' + window.location.host + window.location.pathname+ '/test');
```
Comment (bogensberger) [inline app/tests/services/sql.tests.js]: add space before `+`

Comment (bogensberger) [inline app/tests/services/sql.tests.js]: as it's a test, isn't it possible to compare it with a hardcoded string like `http://example.com/test`. That would make the test more readable.

Comment (m-kharbat) [inline app/tests/services/sql.tests.js]: The baseURI service gets the url directly from the global object window.location I couldn't find a way to override or mock this object without reloading the page

File: CHANGES.txt [hunk 5]
```diff
@@ -5,6 +5,8 @@ Changes for Crate Admin Interface
 Unreleased
 ==========
 
+ - Added compatibility with future crate versions.
```
Comment (bogensberger) [inline CHANGES.txt]: ... with future crate versions which will serve the admin ui from /admin/ instead of `_plugins...

**Pull request merged**